animated_sprite.cpp

```diff
@@ -0,0 +1,158 @@
+#include "animated_sprite.h"
+#include <iostream>
+#include "constants.h"
+
+AnimatedSprite::AnimatedSprite(std::string sprite_sheet_name,
+                               SDL_Rect first_sprite_coord_location,
+                               SDL_Renderer* renderer,
+                               float time_to_update_sprite,
+                               SDL_Rect destination)
+    : Sprite(sprite_sheet_name, first_sprite_coord_location, renderer),
+      current_animation(AnimationType::kIdle),
+      frame_index_(0),
+      destination_(destination) {
+  animation.time_between_frames = time_to_update_sprite;
+  animation.once = false;
+  animation.sprite_sheet_name = sprite_sheet_name;
+  animation.renderer = renderer;
+  this->SetupAnimations();
+}
+
+bool AnimatedSprite::TimeToUpdateFrame() {
+  // Here
+  SetElapsedTime(SDL_GetTicks());
+  if (elapsed_time_ >= animations_[current_animation].time_between_frames) {
+    return true;
+  }
+  return false;
+}
+
+SDL_Rect AnimatedSprite::GetDestinationRect() {
+  destination_.x = (destination_.x + animations_[current_animation].offsets.x) %
```

Comment:
Make a new issue to separate sprite location on the screen from the concept of the animation.